### ISSUE TITLE
Missing manual attribute option values sort order in GraphQL queries

### ISSUE THREAD

**Issue by idziakjakub** [Smile-SA/elasticsuite#1845]
Manual attribute option values sort order isn't working in GraphQL query field "aggregations".
There is no distinction sorting between count and manual option in class `Smile\ElasticsuiteCore\Search\Adapter\Elasticsuite\Request\Aggregation\Builder\Term`:

https://github.com/Smile-SA/elasticsuite/blob/124f01dfa13d12957b07c6541ea44c920069f2d8/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Request/Aggregation/Builder/Term.php#L45-L49

I found method in class `Smile\ElasticsuiteCatalog\Model\Layer\Filter::addOptionsData` responsible for custom sort, but it's only used on frontend:

https://github.com/Smile-SA/elasticsuite/blob/124f01dfa13d12957b07c6541ea44c920069f2d8/src/module-elasticsuite-catalog/Model/Layer/Filter/Attribute.php#L207-L230

### Preconditions

Magento Version : 2.3.4 CE

ElasticSuite Version : 2.8.5

Environment : Developer

Third party modules : 

        "aoepeople/zettr": "1.0.0",
        "macopedia/magmi2": "dev-magento2",
        "olegkoval/magento2-regenerate-url-rewrites": "^1.4",
        "snowdog/theme-blank-sass": "1.3.0",
        "snowdog/frontools": "1.7.3",
        "snowdog/module-menu": "^2.8",
        "staempfli/magento2-module-image-resizer": "2.0.4",
        "mageplaza/magento-2-swedish-language-pack": "dev-newsletter-translations",
        "mageplaza/magento-2-social-login": "^2.8",
        "yireo/magento2-linkpreload": "1.4.0",
        "zf1/zend-reflection": "^1.12",
        "magefan/module-wysiwyg-advanced": "^2.0",
        "creatuity/magento2-interceptors": "dev-master",
        "fsw/magento2-error-sieve": "dev-master",
        "fsw/magento2-cron-runner": "dev-master",
        "creativestyle/magesuite": "^5.0.0",
        "regento/magento2-newsletter-api": "1.0.1"

### Steps to reproduce

1. Go to edit page of some "Dropdown" type attribute used in layered navigation
2. Change manually option values sort
3. Go to "Storefront Properties" tab
4. Change "Facet sort order" to "Admin sort"
5. Run GraphQL query:

```
        query getProductFiltersByCategory($categoryIdFilter: FilterEqualTypeInput!) {
            products(filter: { category_id: $categoryIdFilter }) {
                aggregations {
                    label
                    count
                    attribute_code
                    options {
                        label
                        value
                    }
                }
            }
        }
```

Variables:

```
        {
        	"categoryIdFilter": {
        		"eq": {{TEST CATEGORY ID}}
        	}
        }
```

### Expected result

In response attribute options should be sorted like in admin panel.

### Actual result

In response attribute options are sorted by results count.

**Comment by shivnibble** [comment]
@rbayet @romainruaud @idziakjakub any update on this? Suffering from the same problem, and anxious for a resolution.

**Comment by aligent-lturner** [comment]
@romainruaud can this be re-opened? The issue is that the fix in 4c388ad does not take `attribute_set_id` into account early enough. 
https://github.com/Smile-SA/elasticsuite/blob/4c388add34592e41939d457192fd547e15523e92/src/module-elasticsuite-catalog-graph-ql/DataProvider/Product/LayeredNavigation/Builder/Attribute.php#L115-L116
If the attribute code is `attribute_set_id`, then `$attribute` will be undefined, as it will not be loaded from the repository (as it's not an attribute). I believe these two lines should be changed to:
```
if ($attributeCode !== 'attribute_set_id' &&
    $attribute->getFacetSortOrder() == ElasticBucketInterface::SORT_ORDER_MANUAL) {
```
